clcache.py

```diff
@@ -1612,19 +1618,28 @@ def processCompileRequest(cache, compiler, args):
     printOutAndErr(out, err)
     return exitCode
 
+def formBaseCommandLine(cmdLine, sourceFiles):
+    # type: (List[str], List[Tuple[str, str]]) -> List[str]
+    setOfSources = set([sourceFile for sourceFile, _ in sourceFiles])
+    skippedArgs = ('/MP', '/Tc', '/Tp')
+    baseCmdLine = [
+        arg for arg in cmdLine
+        if not (arg in setOfSources or arg.startswith(skippedArgs))
+    ]
```

Comment:
This list comprehension works for cases like `/Tcfoo.c` but it's also possible to have an optional(!) space, i.e. `/Tc foo.c`. In this case _both_ elements should get removed (since just `/Tc` on its own is a command line error).

Reply:
Ahh, but in the second case, it would be in `setOfSources`.